libs/parsers/src/lib/expression/parser.ts

```diff
@@ -132,11 +139,23 @@ export class ExpressionParser {
       if (scanner.getTokenForTypeAndValue(ExpressionTokenType.LeftSquareBracket)) {
         const expression = this._parseExpressionRecursively(scanner);
         token = ExpressionParser._getTokenOrThrowException(scanner, ExpressionTokenType.RightSquareBracket);
-        dereferences.push({
-          isSafe,
-          isDotNotation: false,
-          expression: expression,
-        });
+
+        if (expression.type === ExpressionType.StringLiteral && isAliasPathParsingEnabled) {
```

Comment:
Are we sure we do not want to do it for ExpressionType stringinterpolation?

Reply:
I'm not sure for this, I'll leave a comment in case we run into issues that indicate string Interpolation should be inlcuded